pkg/compiler/rego/rego.go

```diff
@@ -170,13 +181,20 @@ func (c *CompilerRego) linearizeContext(stmt *ast.ContextStatement) []*ast.Actio
 					}
 				}
 
+				// And append generated ActionStatement to the list
 				line = append(line, cAction)
 			} else {
-				// in case of context in action
+				// in case of context in action it also should be exploded to list of ActionStatement
 				ctx := act.Context
-				for _, icond := range stmt.Conditions {
-					ctx.Conditions = append(ctx.Conditions, icond)
+
+				for _, icond := range stmt.Conditions { // add conditions defined for 'parent' context
+					// but only in case it's not blank, mean parent does not looks like context {}...
+					if icond.Subject != nil || icond.Where != nil {
+						ctx.Conditions = append(ctx.Conditions, icond)
+					}
 				}
+
```

Comment:
Suggestion (also parser.validateContextStatement should be adjusted):
```
				//if Verb is not defined for child context - set it from parent
				if ctx.Verb == nil {
					ctx.Verb = stmt.Verb
				}
				// and the same for TypePattern
				if ctx.TypePattern == nil {
					ctx.TypePattern = stmt.TypePattern
				}

```

Reply:
let's do that in next pr